File: ine-qr-ios/IneQr/IneResult.swift
import Foundation

/// Decoded INE credential.
///
/// - `json`: 18 biographical fields as a UTF-8 JSON string. Always populated.
/// - `webp`: photo as raw WebP bytes (96×129 px). Empty if no photo present.
public struct IneResult: Equatable, Sendable {
    public let json: String
    public let webp: Data

    public init(json: String, webp: Data) {
        self.json = json
        self.webp = webp
    }
}

File: ine-qr-android/ine-qr/src/main/kotlin/mx/ine/qr/IneResult.kt
package mx.ine.qr

/**
 * Decoded INE credential.
 *
 * @property json   18 biographical fields as a UTF-8 JSON string. Always
 *                  populated on success.
 * @property webp   Photo as raw WebP bytes (96×129 px). Empty if the
 *                  credential carried no photo (rare but possible).
 */
class IneResult(
    val json: String,
    val webp: ByteArray,
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is IneResult) return false
        return json == other.json && webp.contentEquals(other.webp)
    }

    override fun hashCode(): Int = 31 * json.hashCode() + webp.contentHashCode()
}
